Maple2.Server.Game/Manager/HousingManager.cs

```diff
@@ -277,4 +279,39 @@ public class HousingManager {
             db.SavePlotInfo(Home.Indoor);
         }
     }
+
+    public void InitNewHome(string characterName, ExportedUgcMapMetadata? template) {
+        Home.Indoor.Name = characterName;
+        Home.Indoor.ExpiryTime = 32503561200; // Year 2999
+        Home.Message = "Thanks for visiting. Come back soon!";
+        Home.DecorationLevel = 1;
+        Home.Passcode = "*****";
+
+        using GameStorage.Request db = session.GameStorage.Context();
+        if (template is null) {
+            Home.SetArea(10);
+            Home.SetHeight(4);
+
+            db.SaveHome(Home);
+            db.SavePlotInfo(Home.Indoor);
+            return;
+        }
+
+        Home.SetArea(template.IndoorSize[0]);
+        Home.SetHeight(template.IndoorSize[2]);
+
+        List<PlotCube> plotCubes = [];
+        foreach (ExportedUgcMapMetadata.Cube cube in template.Cubes) {
+            PlotCube plotCube = new(cube.ItemId, 0, null) {
+                Position = template.BaseCubePosition + cube.OffsetPosition,
+                Rotation = cube.Rotation
+            };
+
+            plotCubes.Add(plotCube);
+        }
+
+        db.SaveHome(Home);
+        db.SavePlotInfo(Home.Indoor);
+        db.SaveCubes(Home.Indoor, plotCubes);
```

Comment:
jw, have you looked at a sniff on when first loading a home? I think what it's doing is properly adding it to the player's furnishing inventory then placing it. This would mean each item is safely added which I don't think we currently have a method of rerouting block item ids to the furnishing inventory yet. Might be worthwhile to look as it would also serve use for obtaining housing blocks from rewards, boxes, etc